### PR TITLE
EVG-1965 make services check admin flags

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: alerts/alerts.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
```
Comment (bsamek) [inline alerts/alerts.go]: Make sure to include a commit message summary (currently it just says "EVG-1965").

File: alerts/alerts.go [hunk 2]
```diff
@@ -273,6 +274,20 @@ func (qp *QueueProcessor) Deliver(req *alert.AlertRequest, ctx *AlertContext) er
...
 func (qp *QueueProcessor) Run(config *evergreen.Settings) error {
 	startTime := time.Now()
+	adminSettings, err := admin.GetSettingsFromDB()
+	if err != nil {
+		grip.Error(errors.Wrap(err, "error retrieving admin settings"))
+	}
+	if adminSettings != nil && adminSettings.ServiceFlags.AlertsDisabled {
+		grip.Info(message.Fields{
+			"runner":  qp.Name(),
+			"status":  "success",
```
Comment (bsamek) [inline alerts/alerts.go]: Is "success" really correct here? Should it be "disabled" or something else more illuminating?

File: alerts/alerts.go [hunk 3]
```diff
@@ -273,6 +274,20 @@ func (qp *QueueProcessor) Deliver(req *alert.AlertRequest, ctx *AlertContext) er
...
+			"runtime": time.Since(startTime),
```
Comment (bsamek) [inline alerts/alerts.go]: Are runtime or span useful here?

**Comment by john-m-liu** [comment]
Addressed comments and also made services no-op if retrieving settings fails

File: alerts/alerts.go [hunk 4]
```diff
@@ -273,6 +274,18 @@ func (qp *QueueProcessor) Deliver(req *alert.AlertRequest, ctx *AlertContext) er
 // Run loops while there are any unprocessed alerts and attempts to deliver them.
 func (qp *QueueProcessor) Run(config *evergreen.Settings) error {
 	startTime := time.Now()
+	adminSettings, err := admin.GetSettingsFromDB()
+	if err != nil {
+		return errors.Wrap(err, "error retrieving admin settings")
+	}
+	if adminSettings != nil && adminSettings.ServiceFlags.AlertsDisabled {
```
Comment (tychoish) [inline alerts/alerts.go]: there should be no situation where adminSettings is nil if there is no error

File: model/admin/db.go [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+	NotificationsKey = bsonutil.MustHaveTag(ServiceFlags{}, "NotificationsDisabled")
+	AlertsKey        = bsonutil.MustHaveTag(ServiceFlags{}, "AlertsDisabled")
+	TaskrunnerKey    = bsonutil.MustHaveTag(ServiceFlags{}, "TaskrunnerDisabled")
+	RepotrackerKey   = bsonutil.MustHaveTag(ServiceFlags{}, "RepotrackerDisabled")
+	SchedulerKey     = bsonutil.MustHaveTag(ServiceFlags{}, "SchedulerDisabled")
+)
+
+var settingsQuery = db.Query(bson.M{IdKey: systemSettingsID})
+
+func GetSettingsFromDB() (*AdminSettings, error) {
```
Comment (tychoish) [inline model/admin/db.go]: "FromDB" seems like its redundantish,

File: monitor/runner.go [hunk 6]
```diff
@@ -27,6 +28,19 @@ func (r *Runner) Description() string {
 
 func (r *Runner) Run(config *evergreen.Settings) error {
 	startTime := time.Now()
+	adminSettings, err := admin.GetSettingsFromDB()
+	if err != nil {
+		return errors.Wrap(err, "error retrieving admin settings")
+	}
+	if adminSettings != nil && adminSettings.ServiceFlags.MonitorDisabled {
```
Comment (tychoish) [inline monitor/runner.go]: nil check should be unneeded

File: model/admin/db.go [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+import (
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/db/bsonutil"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+const (
+	Collection       = "admin"
+	systemSettingsID = "0"
```
Comment (tychoish) [inline model/admin/db.go]: I think we could/should have a name that is a little more explicit than 0 for this key.

File: model/admin/db.go [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+	ServiceFlagsKey  = bsonutil.MustHaveTag(AdminSettings{}, "ServiceFlags")
+	TaskDispatchKey  = bsonutil.MustHaveTag(ServiceFlags{}, "TaskDispatchDisabled")
+	HostinitKey      = bsonutil.MustHaveTag(ServiceFlags{}, "HostinitDisabled")
+	MonitorKey       = bsonutil.MustHaveTag(ServiceFlags{}, "MonitorDisabled")
+	NotificationsKey = bsonutil.MustHaveTag(ServiceFlags{}, "NotificationsDisabled")
+	AlertsKey        = bsonutil.MustHaveTag(ServiceFlags{}, "AlertsDisabled")
+	TaskrunnerKey    = bsonutil.MustHaveTag(ServiceFlags{}, "TaskrunnerDisabled")
+	RepotrackerKey   = bsonutil.MustHaveTag(ServiceFlags{}, "RepotrackerDisabled")
+	SchedulerKey     = bsonutil.MustHaveTag(ServiceFlags{}, "SchedulerDisabled")
```
Comment (tychoish) [inline model/admin/db.go]: Is there a way that we can make the key names private? I don't like the key name paradigm in the other model packages: it means we basically export *everything* and I don't think that's a good pattern to follow.

File: model/admin/db.go [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+		settingsQuery,
+		bson.M{
+			"$set": bson.M{IdKey: systemSettingsID, BannerKey: bannerText},
+		},
+	)
+
+	return err
+}
+
+func SetTaskDispatchDisabled(disableTasks bool) error {
```
Comment (tychoish) [inline model/admin/db.go]: can this be "DisableTaskDispatch"

File: model/admin/db.go [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+		settingsQuery,
+		bson.M{
+			"$set": bson.M{IdKey: systemSettingsID, ServiceFlagsKey: flags},
+		},
+	)
+
+	return err
+}
+
+func Upsert(settings *AdminSettings) error {
```
Comment (tychoish) [inline model/admin/db.go]: does this need to be exported?

is it used?

File: alerts/alerts.go (hunk 4 above)
Comment (john-m-liu) [inline alerts/alerts.go]: Done

File: alerts/alerts.go (hunk 2 above)
Comment (john-m-liu) [inline alerts/alerts.go]: Removed

File: alerts/alerts.go (hunk 3 above)
Comment (john-m-liu) [inline alerts/alerts.go]: Removed

File: model/admin/db.go (hunk 7 above)
Comment (john-m-liu) [inline model/admin/db.go]: Done, but in the EVG-1964 branch

File: model/admin/db.go (hunk 8 above)
Comment (john-m-liu) [inline model/admin/db.go]: Done, but in the EVG-1964 branch

File: model/admin/db.go (hunk 5 above)
Comment (john-m-liu) [inline model/admin/db.go]: Done, but in the EVG-1964 branch

File: model/admin/db.go (hunk 9 above)
Comment (john-m-liu) [inline model/admin/db.go]: This actually was removed in the EVG-1964 branch because it's now redundant

File: model/admin/db.go (hunk 10 above)
Comment (john-m-liu) [inline model/admin/db.go]: Not yet but I was planning on using it in the routes. If I don't end up using it in a route I'll delete it with its test

File: monitor/runner.go (hunk 6 above)
Comment (john-m-liu) [inline monitor/runner.go]: Done

File: model/admin/db.go (hunk 10 above)
Comment (tychoish) [inline model/admin/db.go]: I think the routes should use specific methods "unset/set" values/methods (which will be more atomic) than exposing an upsert/save method?

**Pull request closed without merging**